### PR TITLE
run integration test in openshift

### PR DESCRIPTION
**Pull request by bhavaniravichandran**
jenkins job that ran full integration test. http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/2734/  There was 1 failure and I reran that one "It" class and the test passed. Seems intermittent. Results below
http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/2744/

### CODE REVIEW

**Comment by bhavaniravichandran** [comment]
Don't know why the committer is listed as Mark. I gave my id when I pushed to this branch.

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java [hunk 1]
```diff
@@ -828,21 +829,42 @@ public void createDomainOnExistingDirectory() throws Exception {
...
-    // created on the pv dir
-    try {
-      callCreateDomainScript(userProjectsDir + "2");
-    } catch (RuntimeException re) {
-      re.printStackTrace();
-      logger.info("[SUCCESS] create domain job failed, this is the expected behavior");
-      return;
+      logger.info("making sure the domain directory exists by running " + cmd);
+      ExecResult result = TestUtils.exec(cmd);
+      // logger.info("Command result " + result.stdout() + " err =" + result.stderr());
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java]: else part can start here as rest of the code is same in both conditions..? 

this is same in both conditions ...
```
logger.info("Run the script to create domain");

      // create domain using different output dir but pv is same, it fails as the domain was already
      // created on the pv dir
      try {
        callCreateDomainScript(userProjectsDir + "2");
      } catch (RuntimeException re) {
        re.printStackTrace();
        logger.info("[SUCCESS] create domain job failed, this is the expected behavior");
        return;
      }
throw new RuntimeException("FAIL: unexpected result, create domain job did not report error");
```

File: integration-tests/src/test/resources/statedump.sh [hunk 2]
```diff
@@ -162,6 +183,8 @@ export SCRIPTPATH="$( cd "$(dirname "$0")" > /dev/null 2>&1 ; pwd -P )"
 export PROJECT_ROOT="$SCRIPTPATH/../../../.."
 export RESULT_ROOT=${RESULT_ROOT:-/scratch/$USER/wl_k8s_test_results}
 export PV_ROOT=${PV_ROOT:-$RESULT_ROOT}
+#export OPENSHIFT= ${OPENSHIFT:-false}
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/statedump.sh]: delete

File: integration-tests/src/test/resources/statedump.sh [hunk 3]
```diff
@@ -162,6 +183,8 @@ export SCRIPTPATH="$( cd "$(dirname "$0")" > /dev/null 2>&1 ; pwd -P )"
...
 echo "RESULT_ROOT$RESULT_ROOT PV_ROOT$PV_ROOT"
+#echo "OPENSHIFT$OPENSHIFT"
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/statedump.sh]: delete

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java [hunk 4]
```diff
@@ -1200,23 +1215,38 @@ public static void buildWdtZip(
...
-        + dirPath
-        + "'";
-    
-    ExecResult result = ExecCommand.exec(crdCmd);
-    if (result.exitValue() != 0) {
-      throw new RuntimeException(
+    if (BaseTest.OPENSHIFT) {
+      String crdCmd = "mkdir -m 777 -p " +  dirPath;
+  
+      ExecResult result = ExecCommand.exec(crdCmd);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java]: replace these lines with TestUtils.exec(str, true)

File: integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java [hunk 5]
```diff
@@ -1200,23 +1215,38 @@ public static void buildWdtZip(
...
+      logger.info("command result " + result.stdout().trim());
+    } else {
+      dirPath = dirPath.replace(BaseTest.getPvRoot(), "/sharedparent/");
+      String crdCmd =
+          BaseTest.getProjectRoot()
+          + "/src/integration-tests/bash/krun.sh -m " + BaseTest.getPvRoot() + ":/sharedparent -c 'mkdir -m 777 -p "
+          + dirPath
+          + "'";
+      
+      ExecResult result = ExecCommand.exec(crdCmd);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/utils/TestUtils.java]: as you are changing this code, please do the same here ...replace these lines with TestUtils.exec(str, true)

**Comment by bhavaniravichandran** [comment]
jenkins job passed after my latest push
http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/2768/

**Pull request merged**